packager/media/formats/webm/webm_video_client.cc

```diff
@@ -161,11 +161,11 @@ VPCodecConfigurationRecord WebMVideoClient::GetVpCodecConfig(
 }
 
 WebMParserClient* WebMVideoClient::OnListStart(int id) {
-  return id == kWebMIdColor ? this : WebMParserClient::OnListStart(id);
+  return id == kWebMIdColor || id == kWebMIdProjection ? this : WebMParserClient::OnListStart(id);
```

Comment:
Can you format it so the total line length is within 80 characters? You can use `git clang-format <base commit>` to format the code automatically. https://github.com/google/shaka-packager/blob/master/packager/tools/git/check_formatting.py#L31

Same below.

Reply:
Ok, no problem! maybe i did something wrong but the diff longs quite strange and not like surrunding code:
```sh
$ packager/tools/git/check_formatting.py HEAD~1
diff --git a/packager/media/formats/webm/webm_parser.cc b/packager/media/formats/webm/webm_parser.cc
index d0c8bd3357..677c390b6f 100644
--- a/packager/media/formats/webm/webm_parser.cc
+++ b/packager/media/formats/webm/webm_parser.cc
@@ -182,23 +182,15 @@ static const ElementIdInfo kTrackTranslateIds[] = {
 };

 static const ElementIdInfo kVideoIds[] = {
-  {UINT, kWebMIdFlagInterlaced},
-  {UINT, kWebMIdStereoMode},
-  {UINT, kWebMIdAlphaMode},
-  {UINT, kWebMIdPixelWidth},
-  {UINT, kWebMIdPixelHeight},
-  {UINT, kWebMIdPixelCropBottom},
-  {UINT, kWebMIdPixelCropTop},
-  {UINT, kWebMIdPixelCropLeft},
-  {UINT, kWebMIdPixelCropRight},
-  {UINT, kWebMIdDisplayWidth},
-  {UINT, kWebMIdDisplayHeight},
-  {UINT, kWebMIdDisplayUnit},
-  {UINT, kWebMIdAspectRatioType},
-  {BINARY, kWebMIdColorSpace},
-  {FLOAT, kWebMIdFrameRate},
-  {LIST, kWebMIdColor},
-  {LIST, kWebMIdProjection},
+    {UINT, kWebMIdFlagInterlaced},  {UINT, kWebMIdStereoMode},
+    {UINT, kWebMIdAlphaMode},       {UINT, kWebMIdPixelWidth},
+    {UINT, kWebMIdPixelHeight},     {UINT, kWebMIdPixelCropBottom},
+    {UINT, kWebMIdPixelCropTop},    {UINT, kWebMIdPixelCropLeft},
+    {UINT, kWebMIdPixelCropRight},  {UINT, kWebMIdDisplayWidth},
+    {UINT, kWebMIdDisplayHeight},   {UINT, kWebMIdDisplayUnit},
+    {UINT, kWebMIdAspectRatioType}, {BINARY, kWebMIdColorSpace},
+    {FLOAT, kWebMIdFrameRate},      {LIST, kWebMIdColor},
+    {LIST, kWebMIdProjection},
 };

 static const ElementIdInfo kColorIds[] = {
@@ -219,7 +211,7 @@ static const ElementIdInfo kColorIds[] = {
 };

 static const ElementIdInfo kProjectionIds[] = {
-  {UINT, kWebMIdProjectionType},
+    {UINT, kWebMIdProjectionType},
 };

 static const ElementIdInfo kAudioIds[] = {
@@ -388,52 +380,54 @@ static const ElementIdInfo kSimpleTagIds[] = {
     { (id), (level), (id_info), arraysize(id_info) }

 static const ListElementInfo kListElementInfo[] = {
-  LIST_ELEMENT_INFO(kWebMIdCluster, 1, kClusterIds),
-  LIST_ELEMENT_INFO(kWebMIdEBMLHeader, 0, kEBMLHeaderIds),
-  LIST_ELEMENT_INFO(kWebMIdSegment, 0, kSegmentIds),
-  LIST_ELEMENT_INFO(kWebMIdSeekHead, 1, kSeekHeadIds),
-  LIST_ELEMENT_INFO(kWebMIdSeek, 2, kSeekIds),
-  LIST_ELEMENT_INFO(kWebMIdInfo, 1, kInfoIds),
-  LIST_ELEMENT_INFO(kWebMIdChapterTranslate, 2, kChapterTranslateIds),
-  LIST_ELEMENT_INFO(kWebMIdSilentTracks, 2, kSilentTracksIds),
-  LIST_ELEMENT_INFO(kWebMIdBlockGroup, 2, kBlockGroupIds),
-  LIST_ELEMENT_INFO(kWebMIdBlockAdditions, 3, kBlockAdditionsIds),
-  LIST_ELEMENT_INFO(kWebMIdBlockMore, 4, kBlockMoreIds),
-  LIST_ELEMENT_INFO(kWebMIdSlices, 3, kSlicesIds),
-  LIST_ELEMENT_INFO(kWebMIdTimeSlice, 4, kTimeSliceIds),
-  LIST_ELEMENT_INFO(kWebMIdTracks, 1, kTracksIds),
-  LIST_ELEMENT_INFO(kWebMIdTrackEntry, 2, kTrackEntryIds),
-  LIST_ELEMENT_INFO(kWebMIdTrackTranslate, 3, kTrackTranslateIds),
-  LIST_ELEMENT_INFO(kWebMIdVideo, 3, kVideoIds),
-  LIST_ELEMENT_INFO(kWebMIdColor, 4, kColorIds),
-  LIST_ELEMENT_INFO(kWebMIdProjection, 4, kProjectionIds),
-  LIST_ELEMENT_INFO(kWebMIdAudio, 3, kAudioIds),
-  LIST_ELEMENT_INFO(kWebMIdTrackOperation, 3, kTrackOperationIds),
-  LIST_ELEMENT_INFO(kWebMIdTrackCombinePlanes, 4, kTrackCombinePlanesIds),
-  LIST_ELEMENT_INFO(kWebMIdTrackPlane, 5, kTrackPlaneIds),
-  LIST_ELEMENT_INFO(kWebMIdJoinBlocks, 4, kJoinBlocksIds),
-  LIST_ELEMENT_INFO(kWebMIdContentEncodings, 3, kContentEncodingsIds),
-  LIST_ELEMENT_INFO(kWebMIdContentEncoding, 4, kContentEncodingIds),
-  LIST_ELEMENT_INFO(kWebMIdContentCompression, 5, kContentCompressionIds),
-  LIST_ELEMENT_INFO(kWebMIdContentEncryption, 5, kContentEncryptionIds),
-  LIST_ELEMENT_INFO(kWebMIdContentEncAESSettings, 6, kContentEncAESSettingsIds),
-  LIST_ELEMENT_INFO(kWebMIdCues, 1, kCuesIds),
-  LIST_ELEMENT_INFO(kWebMIdCuePoint, 2, kCuePointIds),
-  LIST_ELEMENT_INFO(kWebMIdCueTrackPositions, 3, kCueTrackPositionsIds),
-  LIST_ELEMENT_INFO(kWebMIdCueReference, 4, kCueReferenceIds),
-  LIST_ELEMENT_INFO(kWebMIdAttachments, 1, kAttachmentsIds),
-  LIST_ELEMENT_INFO(kWebMIdAttachedFile, 2, kAttachedFileIds),
-  LIST_ELEMENT_INFO(kWebMIdChapters, 1, kChaptersIds),
-  LIST_ELEMENT_INFO(kWebMIdEditionEntry, 2, kEditionEntryIds),
-  LIST_ELEMENT_INFO(kWebMIdChapterAtom, 3, kChapterAtomIds),
-  LIST_ELEMENT_INFO(kWebMIdChapterTrack, 4, kChapterTrackIds),
-  LIST_ELEMENT_INFO(kWebMIdChapterDisplay, 4, kChapterDisplayIds),
-  LIST_ELEMENT_INFO(kWebMIdChapProcess, 4, kChapProcessIds),
-  LIST_ELEMENT_INFO(kWebMIdChapProcessCommand, 5, kChapProcessCommandIds),
-  LIST_ELEMENT_INFO(kWebMIdTags, 1, kTagsIds),
-  LIST_ELEMENT_INFO(kWebMIdTag, 2, kTagIds),
-  LIST_ELEMENT_INFO(kWebMIdTargets, 3, kTargetsIds),
-  LIST_ELEMENT_INFO(kWebMIdSimpleTag, 3, kSimpleTagIds),
+    LIST_ELEMENT_INFO(kWebMIdCluster, 1, kClusterIds),
+    LIST_ELEMENT_INFO(kWebMIdEBMLHeader, 0, kEBMLHeaderIds),
+    LIST_ELEMENT_INFO(kWebMIdSegment, 0, kSegmentIds),
+    LIST_ELEMENT_INFO(kWebMIdSeekHead, 1, kSeekHeadIds),
+    LIST_ELEMENT_INFO(kWebMIdSeek, 2, kSeekIds),
+    LIST_ELEMENT_INFO(kWebMIdInfo, 1, kInfoIds),
+    LIST_ELEMENT_INFO(kWebMIdChapterTranslate, 2, kChapterTranslateIds),
+    LIST_ELEMENT_INFO(kWebMIdSilentTracks, 2, kSilentTracksIds),
+    LIST_ELEMENT_INFO(kWebMIdBlockGroup, 2, kBlockGroupIds),
+    LIST_ELEMENT_INFO(kWebMIdBlockAdditions, 3, kBlockAdditionsIds),
+    LIST_ELEMENT_INFO(kWebMIdBlockMore, 4, kBlockMoreIds),
+    LIST_ELEMENT_INFO(kWebMIdSlices, 3, kSlicesIds),
+    LIST_ELEMENT_INFO(kWebMIdTimeSlice, 4, kTimeSliceIds),
+    LIST_ELEMENT_INFO(kWebMIdTracks, 1, kTracksIds),
+    LIST_ELEMENT_INFO(kWebMIdTrackEntry, 2, kTrackEntryIds),
+    LIST_ELEMENT_INFO(kWebMIdTrackTranslate, 3, kTrackTranslateIds),
+    LIST_ELEMENT_INFO(kWebMIdVideo, 3, kVideoIds),
+    LIST_ELEMENT_INFO(kWebMIdColor, 4, kColorIds),
+    LIST_ELEMENT_INFO(kWebMIdProjection, 4, kProjectionIds),
+    LIST_ELEMENT_INFO(kWebMIdAudio, 3, kAudioIds),
+    LIST_ELEMENT_INFO(kWebMIdTrackOperation, 3, kTrackOperationIds),
+    LIST_ELEMENT_INFO(kWebMIdTrackCombinePlanes, 4, kTrackCombinePlanesIds),
+    LIST_ELEMENT_INFO(kWebMIdTrackPlane, 5, kTrackPlaneIds),
+    LIST_ELEMENT_INFO(kWebMIdJoinBlocks, 4, kJoinBlocksIds),
+    LIST_ELEMENT_INFO(kWebMIdContentEncodings, 3, kContentEncodingsIds),
+    LIST_ELEMENT_INFO(kWebMIdContentEncoding, 4, kContentEncodingIds),
+    LIST_ELEMENT_INFO(kWebMIdContentCompression, 5, kContentCompressionIds),
+    LIST_ELEMENT_INFO(kWebMIdContentEncryption, 5, kContentEncryptionIds),
+    LIST_ELEMENT_INFO(kWebMIdContentEncAESSettings,
+                      6,
+                      kContentEncAESSettingsIds),
+    LIST_ELEMENT_INFO(kWebMIdCues, 1, kCuesIds),
+    LIST_ELEMENT_INFO(kWebMIdCuePoint, 2, kCuePointIds),
+    LIST_ELEMENT_INFO(kWebMIdCueTrackPositions, 3, kCueTrackPositionsIds),
+    LIST_ELEMENT_INFO(kWebMIdCueReference, 4, kCueReferenceIds),
+    LIST_ELEMENT_INFO(kWebMIdAttachments, 1, kAttachmentsIds),
+    LIST_ELEMENT_INFO(kWebMIdAttachedFile, 2, kAttachedFileIds),
+    LIST_ELEMENT_INFO(kWebMIdChapters, 1, kChaptersIds),
+    LIST_ELEMENT_INFO(kWebMIdEditionEntry, 2, kEditionEntryIds),
+    LIST_ELEMENT_INFO(kWebMIdChapterAtom, 3, kChapterAtomIds),
+    LIST_ELEMENT_INFO(kWebMIdChapterTrack, 4, kChapterTrackIds),
+    LIST_ELEMENT_INFO(kWebMIdChapterDisplay, 4, kChapterDisplayIds),
+    LIST_ELEMENT_INFO(kWebMIdChapProcess, 4, kChapProcessIds),
+    LIST_ELEMENT_INFO(kWebMIdChapProcessCommand, 5, kChapProcessCommandIds),
+    LIST_ELEMENT_INFO(kWebMIdTags, 1, kTagsIds),
+    LIST_ELEMENT_INFO(kWebMIdTag, 2, kTagIds),
+    LIST_ELEMENT_INFO(kWebMIdTargets, 3, kTargetsIds),
+    LIST_ELEMENT_INFO(kWebMIdSimpleTag, 3, kSimpleTagIds),
 };

 // Parses an element header id or size field. These fields are variable length
diff --git a/packager/media/formats/webm/webm_video_client.cc b/packager/media/formats/webm/webm_video_client.cc
index edd46d4214..48486a5a55 100644
--- a/packager/media/formats/webm/webm_video_client.cc
+++ b/packager/media/formats/webm/webm_video_client.cc
@@ -161,11 +161,15 @@ VPCodecConfigurationRecord WebMVideoClient::GetVpCodecConfig(
 }

 WebMParserClient* WebMVideoClient::OnListStart(int id) {
-  return id == kWebMIdColor || id == kWebMIdProjection ? this : WebMParserClient::OnListStart(id);
+  return id == kWebMIdColor || id == kWebMIdProjection
+             ? this
+             : WebMParserClient::OnListStart(id);
 }

 bool WebMVideoClient::OnListEnd(int id) {
-  return id == kWebMIdColor || id == kWebMIdProjection ? true : WebMParserClient::OnListEnd(id);
+  return id == kWebMIdColor || id == kWebMIdProjection
+             ? true
+             : WebMParserClient::OnListEnd(id);
 }

 bool WebMVideoClient::OnUInt(int id, int64_t val) {


Code style is not correct. Please run git clang-format --style Chromium HEAD~1.

```

Reply:
No. There is nothing wrong in what you did. 

It is a problem in the original code :(.

Can you just change the formatting for `OnListStart` and `OnListEnd`?

You are welcome to submit a follow-up PR later to fix the formatting in this file.

Reply:
Ok hope i got i right now

Reply:
> Thanks for the pull request! I think that is good enough for now.
> 
> > I guess i should add some tests also
> 
> As long as you have tested and verified that it is working for your test content, I think we can skip the unit-tests for this code as it is not too complicated.

Yes have test a few and the test files in included in issue:

```sh
# ./src/out/Debug/packager  'input=test_projection.mkv,stream=video,format=webm,init_segment=init.webm,segment_template=$Number$.webm'
[0507/090841:INFO:demuxer.cc(89)] Demuxer::Run() on file 'test_projection.mkv'.
[0507/090841:INFO:demuxer.cc(155)] Initialize Demuxer for file 'test_projection.mkv'.
[0507/090841:WARNING:webm_video_client.cc(241)] Ignoring ProjectionType with value 1
[0507/090841:INFO:webm_muxer.cc(58)] WEBM file 'init.webm' finalized.
Packaging completed successfully.
# ./src/out/Debug/packager 'input=test_no_projection.mkv,stream=video,format=webm,init_segment=init.webm,segment_template=$Number$.webm'
[0507/090847:INFO:demuxer.cc(89)] Demuxer::Run() on file 'test_no_projection.mkv'.
[0507/090847:INFO:demuxer.cc(155)] Initialize Demuxer for file 'test_no_projection.mkv'.
[0507/090847:WARNING:vp_codec_configuration_record.cc(45)] VPx video full range flag is inconsistent, 1 vs 0
[0507/090847:INFO:webm_muxer.cc(58)] WEBM file 'init.webm' finalized.
Packaging completed successfully.
```